docs/src/userguide/numpy_tutorial.rst

```diff
@@ -472,21 +367,16 @@ here has gone through a lot of thinking and planning already; still they may
 or may not happen depending on available developer time and resources for
 Cython.
 
-1. Support for efficient access to structs/records stored in arrays; currently
-   only primitive types are allowed.
-2. Support for efficient access to complex floating point types in arrays. The
+1. Support for efficient access to complex floating point types in arrays. The
    main obstacle here is getting support for efficient complex datatypes in
    Cython.
-3. Calling NumPy/SciPy functions currently has a Python call overhead; it
+2. Calling NumPy/SciPy functions currently has a Python call overhead; it
    would be possible to take a short-cut from Cython directly to C. (This does
    however require some isolated and incremental changes to those libraries;
    mail the Cython mailing list for details).
-4. Efficient code that is generic with respect to the number of dimensions.
+3. Efficient code that is generic with respect to the number of dimensions.
    This can probably be done today by calling the NumPy C multi-dimensional
    iterator API directly; however it would be nice to have for-loops over
    :func:`enumerate` and :func:`ndenumerate` on NumPy arrays create efficient
    code.
-5. A high-level construct for writing type-generic code, so that one can write
-   functions that work simultaneously with many datatypes. Note however that a
-   macro preprocessor language can help with doing this for now.
 
```

Comment:
Since there's almost nothing left in this "future" section, I'd vote for deleting it.